subprojects/core/src/main/java/org/gradle/internal/filewatch/jdk7/WatchServiceFileWatcherBacking.java

```diff
@@ -91,7 +91,7 @@ public void run() {
                             try {
                                 pumpEvents();
                             } catch (InterruptedException e) {
-                                // just stop
+                                Thread.currentThread().interrupt();
```

Comment:
If we had that `Interrupts` class I mentioned above, we could have `Interrupts.interruptWithoutThrowing(e)` here. Might not make matters better, though.